x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/DateUtils.java

```diff
@@ -6,19 +6,40 @@
 
 package org.elasticsearch.xpack.ql.type;
 
-import org.elasticsearch.common.time.DateFormatter;
 import org.elasticsearch.common.time.DateFormatters;
 
 import java.time.Instant;
 import java.time.ZoneId;
 import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_TIME;
 
 //NB: Taken from sql-proto.
 public final class DateUtils {
 
     public static final ZoneId UTC = ZoneId.of("Z");
 
-    private static final DateFormatter UTC_DATE_TIME_FORMATTER = DateFormatter.forPattern("date_optional_time").withZone(UTC);
+    private static final DateTimeFormatter DATE_OPTIONAL_TIME_FORMATTER_WHITESPACE = new DateTimeFormatterBuilder()
```

Comment:
Why are these needed in QL and not SQL only?
Also, SQL has a `DateUtils` class with DateTimeFormatters that use whitespace separators. Haven't looked into this in more depth, but have you considered having a single definition source for these DateTimeFormatters?

Reply:
Currently SQL is using the QL for the String->DATETIME conversion, and SQL has the additional conversions for String->DATE, String->TIME. If we move the leniency only to SQL we'll have to move that conversion there. What do you think?

Also we have this issue for consolidating the date related utils: #56494 so I'd rather do the whole refactoring in a separate PR.